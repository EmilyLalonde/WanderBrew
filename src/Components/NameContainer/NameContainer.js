import React from 'react'
import { connect } from 'react-redux';
import './NameContainer.css'
import Card from '../Card/Card'

const NameContainer = ({nameResults}) => {
  let searchResults = nameResults.map(nameObj => {
      return <Card 
      name={nameObj.name}
      street={nameObj.street} 
      city={nameObj.city}
      state={nameObj.state}
      phone={nameObj.phone}
      key={nameObj.id}
      id={nameObj.id}
      />
  })
  return (
    <div>
      {searchResults}
    </div>
  )
}

const mapStateToProps = state => ({
  nameResults:state.nameResults
})

export default connect(mapStateToProps, null)(NameContainer)