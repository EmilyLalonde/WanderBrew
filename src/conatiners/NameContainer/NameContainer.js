import React from 'react'
import { connect } from 'react-redux';
import { Link } from 'react-router-dom'
import './NameContainer.css'
import Card from '../../Components/Card/Card'

const NameContainer = ({nameResults}) => {
  let searchResults = nameResults.map(nameObj => {
      return (
      <Link to={`/name-search/${nameObj.id}`} key={nameObj.id}>
      <Card 
      name={nameObj.name}
      street={nameObj.street} 
      city={nameObj.city}
      state={nameObj.state}
      phone={nameObj.phone}
      website={nameObj.website_url}
      key={nameObj.id}
      id={nameObj.id}
      />
      </Link>
      )
  })
  return (
    <div className="name-search-results">
      {searchResults}
    </div>
  )
}

const mapStateToProps = state => ({
  nameResults:state.nameResults
})

export default connect(mapStateToProps, null)(NameContainer)