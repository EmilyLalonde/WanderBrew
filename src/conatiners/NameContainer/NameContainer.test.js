import React from 'react'
import { shallow } from 'enzyme'
import {NameContainer} from './NameContainer'

describe('NameContainer', () => {
  let wrapper;
  let mockBreweries;

  beforeEach(() => {
    mockBreweries = [{id: 1516, name: 'Ratio Beerworks', street: '2920 Larimer St', city: 'Denver', state:'Colorado', phone:'3012669264'}]
    wrapper= shallow(<NameContainer nameResults={mockBreweries}/>)
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})