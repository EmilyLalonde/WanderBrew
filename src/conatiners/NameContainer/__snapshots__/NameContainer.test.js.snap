// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameContainer should match snapshot 1`] = `
<div
  className="name-search-results"
>
  <Link
    key="1516"
    to="/name-search/1516"
  >
    <Card
      city="Denver"
      id={1516}
      key="1516"
      name="Ratio Beerworks"
      phone="3012669264"
      state="Colorado"
      street="2920 Larimer St"
    />
  </Link>
</div>
`;
